build/build.c

```diff
@@ -237,11 +343,45 @@ static rpmRC buildSpec(BTA_t buildArgs, rpmSpec spec, int what)
     } else {
 	int didBuild = (what & (RPMBUILD_PREP|RPMBUILD_BUILD|RPMBUILD_INSTALL));
 
+	if (!spec->buildrequires && (what & RPMBUILD_PACKAGESOURCE) &&
+	    !(what & (RPMBUILD_BUILD|RPMBUILD_INSTALL|RPMBUILD_PACKAGEBINARY))){
+		/* don't run prep if not needed for source build */
+		/* with(out) dynamic build requires*/
+	    what &= ~(RPMBUILD_PREP);
+	}
+
+	if ((what & RPMBUILD_CHECKBUILDREQUIRES) &&
```

Comment:
Do we need to check it here? Given that we check it few lines later.

Reply:
Well, these dependencies may be needed during %prep or (more likely) during %generate_buildrequires. We don't want these to just blow up. The later check makes sure the newly created buildrequires are there.